content/articles/compile-your-node.js-application-into-a-.exe-file/index.md

```diff
@@ -0,0 +1,537 @@
+
+- In the second, we can define the resources and others inside the 'package.json' file. I recommend this since your configurations would be saved for later re-run. Since we already declared that our app "main" is "index.js", we shall add only a build script.
+Head over to the "package.json" file and under the scripts tag, add a build script as follows:
+
+```json
+{
+    "scripts": {
+        "start": "node index.js",
+        "build": "nexe -r views/**/*",
+        "test": "echo \"Error: no test specified\" && exit 1"
+    }
+}
+```
+
+This shall specify our resources which are locally found during the build time.
+Now run the command below to tell `nexe` to use the specifications in the build tag:
+
+```bash
+npm run build
+```
+
+then to run the build process we shall use:
+
+```bash
+npm --build
+```
+
+You can run the file created. You can also copy it to another different directory without any resources and try running it to verify that the resources were compiled into the executable file.
+
+In case of further errors, you can post them in the [nexe issues tab](https://github.com/nexe/nexe/issues "nexe issues tab") for help.
+
+****
+
+### pkg
+
+![pkg](/engineering-education/compile-your-node.js-application-into-a-.exe-file/nexe.png "pkg")
+
+Let us use the project previously created and see the outcome.
+
+> You can either delete the contents of the root directory (executable) previously created or create a new one with a different name and repeat the steps we did above to create a new project. This reduces vulnerability errors due to packages clashing with each other.
+
+#### pkg installation
+
+You can start by installing the package using the command below for local installation:
+
+```bash
+npm i pkg
+```
+
+or for global installation use:
+
+```bash
+npm i pkg -g
+```
+
+Wait for a few seconds for it to complete the process. On completion, just check and close any instance in which the code is running in the background.
+
+#### Compile one javascript file into an executable using pkg
+
+Run the following command in which we shall call the pkg module and the entry point to our application. In our situation it is `index.js`:
+
+```bash
+pkg index.js
+```
+
+This will only compile our app entry point into three different executable files since no targets were specified.
+Just wait for some time for the process to complete. Head over to the main directory. You will find three newly generated files. These are the:
+
+- for Windows (index-win.exe)
+- for Linux (index-linux)
+- for mac (index-macos)
+
+That is because we did not specify one O/S.
+Run the file suitable for your O/S.
+If you are using Ubuntu, run `./index-linux` on the terminal.
+It will launch the app and display a console that will act as an interactive interface between you and the app. Access the application in the preferred browser at `localhost:5000`. As seen, it works well.
```

Comment:
```suggestion
That is because we did not specify an operating system. Run the file that's suitable for your operating system. If you are using Ubuntu, run `./index-linux` on the terminal. It will launch the app and display a console that will act as an interactive interface between you and the app. Access the application in the preferred browser at `localhost:5000`. As seen, it works well.
```